snow/engine/snowman/block/state_syncable_vm.go

```diff
@@ -42,4 +48,21 @@ type StateSyncableVM interface {
 	// Returns database.ErrNotFound if no summary is available at
 	// [summaryHeight].
 	GetStateSummary(ctx context.Context, summaryHeight uint64) (StateSummary, error)
+
+	// BackfillBlocksEnabled checks if VM wants to download all blocks from state summary one
+	// down to genesis.
+	//
+	// Returns the ID and height of the block it wants to start backfilling from.
+	// Returns ErrBlockBackfillingNotEnabled if block backfilling is not enabled.
+	// BackfillBlocksEnabled can be called multiple times by the engine
+	BackfillBlocksEnabled(ctx context.Context) (ids.ID, uint64, error)
```

Comment:
design note: blkID is not enough. In ProposerVM I need to map blockID requested by innerVM with blockID at proserVM leve. blkHeight helps with that